File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/IProductTypeRepository.java
package br.org.restaurantepopular.usecase.product_type;

import br.org.restaurantepopular.entity.ProductType;

import java.util.List;
import java.util.Optional;

public interface IProductTypeRepository {

    void save(ProductType productType);

    void update(ProductType productType);

    List<ProductType> listProductTypes();

    Optional<ProductType> findById(String id);

    void delete(String id);
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/IProductTypeRepository.kt
package br.org.restaurantepopular.usecase.product_type

import br.org.restaurantepopular.entity.ProductType

interface IProductTypeRepository {

    fun save(productType: ProductType): ProductType

    fun update(productType: ProductType)

    fun listProductTypes(): List<ProductType>

    fun findById(id: String): ProductType

    fun delete(id: String)
}
